packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -356,7 +356,9 @@ function commitHookEffectListMount(tag: number, finishedWork: Fiber) {
       if ((effect.tag & tag) === tag) {
         // Mount
         const create = effect.create;
-        effect.destroy = create();
+        if (typeof create === 'function') {
```

Comment:
I don't think we should make this a variable type unnecessarily. The VM should be able to check that the "effect" object has a consistent type. The appropriate thing here is probably a bigger refactor but it's not worth introducing this discrepancy for this hook (and potentially making all other effects slower).

I'd prefer that we set this to an empty function in this implementation if we need to rather.

Reply:
> I'd prefer that we set this to an empty function in this implementation if we need to rather.

Interesting. I had considered doing that but this seemed more appropriate.

I don't feel strongly about this though. Happy to make that change.

Reply:
Just in case it's unclear, I don't think this code path is actually executed. The change was only needed currently to satisfy Flow (although that could change in the future of course).

> I'd prefer that we set this to an empty function in this implementation if we need to rather.

I considered doing that initially... Happy to make that change!